libobs/obs-video.c

```diff
@@ -664,22 +710,62 @@ static inline void copy_rgbx_frame(struct video_frame *output,
 }
 
 static inline void output_video_data(struct obs_core_video *video,
-				     struct video_data *input_frame, int count)
+				     struct video_data *main_input_frame,
+				     struct video_data *streaming_input_frame,
+				     struct video_data *recording_input_frame,
+				     int count)
 {
 	const struct video_output_info *info;
-	struct video_frame output_frame;
+	struct video_frame main_output_frame;
+	struct video_frame streaming_output_frame;
+	struct video_frame recording_output_frame;
 	bool locked;
 
 	info = video_output_get_info(video->video);
 
-	locked = video_output_lock_frame(video->video, &output_frame, count,
-					 input_frame->timestamp);
+	if (!obs_get_multiple_rendering()) {
+		locked = video_output_lock_frame(video->video,
+						 &main_output_frame, count,
+						 main_input_frame->timestamp,
+						 OBS_MAIN_VIDEO_RENDERING);
+	} else {
+		locked = video_output_lock_frame(
+			video->video, &streaming_output_frame, count,
+			streaming_input_frame->timestamp,
+			OBS_STREAMING_VIDEO_RENDERING);
+		locked = video_output_lock_frame(
```

Comment:
can be **locked &= video_output_lock_frame(** to have kind of correct value in locked flag for both outputs.

Reply:
Good catch, I actually fixed it in another sub branch I have, forgot to put it in here.